pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -88,11 +94,27 @@ type cloudWatchExecutor struct {
 	rgtaClient resourcegroupstaggingapiiface.ResourceGroupsTaggingAPIAPI
 
 	logsService *LogsService
+	settings    *awsSettings
 }
 
 func (e *cloudWatchExecutor) newSession(region string) (*session.Session, error) {
 	dsInfo := e.getDSInfo(region)
 
+	authTypeAllowed := false
+	for _, provider := range e.settings.AllowedAuthProviders {
+		if provider == dsInfo.AuthType.String() {
+			authTypeAllowed = true
+			break
+		}
+	}
+	if !authTypeAllowed {
+		return nil, fmt.Errorf("attempting to use an auth type that is not allowed: %s", dsInfo.AuthType.String())
+	}
+
+	if dsInfo.AssumeRoleARN != "" && !e.settings.AssumeRoleEnabled {
+		plog.Warn("Attempting to use assume role (ARN) which is disabled in grafana.ini. Assume role field will be ignored")
```

Comment:
Assuming a role in a cloud environment won't work, so this is something that was requested specifically by HG. Even though assuming a role won't work in HG, there was nothing stopping users from adding something to the ARN field. For that reason, I didn't want to panic or return an error in the scenario, but rather just log a warning and ignore the assume role. 

Thoughts on that @aknuds1 and @bergquist?

Reply:
I was wondering if this should also be an error since we can't do what the user has requested (assuming a role).

Can't we disable the role field if role assumption is disabled on the server?

Reply:
If assume role is disabled in the grafana.ini, the field will be removed from the UI. But it could be the case that the assume role field was filled in before it was disabled on the server. Then it will stop working. But I guess that's what you want. I'll return an error.

Reply:
> If assume role is disabled in the grafana.ini, the field will be removed from the UI. But it could be the case that the assume role field was filled in before it was disabled on the server. Then it will stop working.

That's a good point, didn't think of that one. I'm not 100% certain, but it seems to me the user would want an error, since the requested role assumption isn't possible.